app/desktop/studio_server/provider_api.py

```diff
@@ -1127,6 +1153,41 @@ async def connect_bedrock(key_data: dict):
     )
 
 
+async def connect_kiln_copilot(key: str):
+    try:
+        base_url = os.environ.get("KILN_SERVER_BASE_URL", "https://api.kiln.tech")
+        # Temporary endpoint to verify the API key
+        response = requests.post(
```

Comment:
never use "requests", always use https. Requests is sync and blocks the whole server.

Related: we could use our nice fancy python API here. But I'm also fine with httpx.